src/NormalDepthMap.cpp

```diff
@@ -73,10 +53,39 @@ bool NormalDepthMap::isDrawNormal() {
 
 void NormalDepthMap::setDrawDepth(bool drawDepth) {
     _drawDepth = drawDepth;
+    if (_normalDepthMapNode != NULL)
+        _normalDepthMapNode->getOrCreateStateSet()->getUniform("drawDepth")->set(_drawDepth);
+}
+
+void NormalDepthMap::addNodeChild(osg::ref_ptr<osg::Node> node) {
+    _normalDepthMapNode->addChild(node);
 }
 
 bool NormalDepthMap::isDrawDepth() {
     return _drawDepth;
 }
 
+void NormalDepthMap::createTheNormalDepthMapShaderNode() {
```

Comment:
OK!!
I will follow your suggestion.
But I tried to concentrate all the methods about the NormalDepthMap, to avoid the necessity to create methods to handle the map, and the node.